packages/babel-plugin-transform-new-target/src/index.ts

```diff
@@ -61,6 +58,23 @@ export default declare(api => {
             return;
           }
 
+          if (!node.id) {
+            node.id = scope.generateUidIdentifier("target");
+          } else {
+            // packages/babel-helper-create-class-features-plugin/src/fields.ts#L192 unshadow
```

Comment:
We could actually move this to a `Scope#unshadow(name, targetAncestorScope)` method, and use it here like
```js
path.scope.unshadow?.(name, func.parentPath.scope);
```

It will only work when people update their `@babel/traverse` (usually by updating their `@babel/core`), but it's not a breaking change because when using older versions `new.target` will just continue to be broken.

Then, we can use `scope.unshadow()` in `babel-helper-create-class-features-plugin` starting from Babel 8 (or better, now with a fallback to the inline `unshadow()` function when `!process.env.BABEL_8_BREAKING`).

Reply:
I feel a little unnecessary.
Because `unshadow` is only a few lines of code and may be different in different scenarios.
Another major reason is that our current Scope is a bit messy, eg binding is not always reliable.
Maybe we can do it in the future.

Reply:
Oh ok fair enough, we can refactor if needed.